mochi-tests repo isn't building due to undefined references to margo_diag_dump() and related functions

mochi-tests fails the compilation with the following error messages.  

make  all-am
make[1]: Entering directory '/lus/theta-fs0/projects/radix-io/vchoi/SCRIPT_TESTING/mochi-tests/BUILD'
  CC       perf-regression/margo-p2p-latency.o
../perf-regression/margo-p2p-latency.c: In function 'main':
../perf-regression/margo-p2p-latency.c:114:32: warning: implicit declaration of function 'margo_diag_start' [-Wimplicit-function-declaration]
  114 |     if (g_opts.diag_file_name) margo_diag_start(mid);
      |                                ^~~~~~~~~~~~~~~~
../perf-regression/margo-p2p-latency.c:205:32: warning: implicit declaration of function 'margo_diag_dump'; did you mean 'margo_state_dump'? [-Wimplicit-function-declaration]
  205 |     if (g_opts.diag_file_name) margo_diag_dump(mid, g_opts.diag_file_name, 1);
      |                                ^~~~~~~~~~~~~~~
      |                                margo_state_dump
  CCLD     perf-regression/margo-p2p-latency
/usr/bin/ld: perf-regression/margo-p2p-latency.o: in function `main':
/lus/theta-fs0/projects/radix-io/vchoi/SCRIPT_TESTING/mochi-tests/BUILD/../perf-regression/margo-p2p-latency.c:114: undefined reference to `margo_diag_start'
/usr/bin/ld: /lus/theta-fs0/projects/radix-io/vchoi/SCRIPT_TESTING/mochi-tests/BUILD/../perf-regression/margo-p2p-latency.c:205: undefined reference to `margo_diag_dump'
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:933: perf-regression/margo-p2p-latency] Error 1

--- a/perf-regression/bake-p2p-bw.c
+++ b/perf-regression/bake-p2p-bw.c
@@ -35,7 +35,6 @@ struct options {
     int           concurrency;
     unsigned int  mercury_timeout_client;
     unsigned int  mercury_timeout_server;
-    char*         diag_file_name;
     char*         na_transport;
     char*         bake_pool;
     int           rpc_xstreams;
@@ -136,8 +135,6 @@ int main(int argc, char** argv)
     /* actually start margo */
     mid = margo_init_ext(g_opts.na_transport, MARGO_SERVER_MODE, &mii);
     assert(mid);
-
-    if (g_opts.diag_file_name) margo_diag_start(mid);
 
     /* adjust mercury timeout in Margo if requested */
     if (my_mpi_rank > 0 && g_opts.mercury_timeout_client != UINT_MAX) {
@@ -272,8 +269,6 @@ int main(int argc, char** argv)
     assert(ret == SSG_SUCCESS);
     ssg_finalize();
 
-    if (g_opts.diag_file_name) margo_diag_dump(mid, g_opts.diag_file_name, 1);
-
     if (my_mpi_rank > 0) free(g_buffer);
     free(gid_buffer);
 
@@ -299,18 +294,11 @@ static int parse_args(int argc, char** argv, struct options* opts)
     opts->mercury_timeout_client = UINT_MAX;
     opts->mercury_timeout_server = UINT_MAX;
 
-    while ((opt = getopt(argc, argv, "n:x:c:d:t:p:m:r:i")) != -1) {
+    while ((opt = getopt(argc, argv, "n:x:c:t:p:m:r:i")) != -1) {
         switch (opt) {
         case 'p':
             opts->bake_pool = strdup(optarg);
             if (!opts->bake_pool) {
-                perror("strdup");
-                return (-1);
-            }
-            break;
-        case 'd':
-            opts->diag_file_name = strdup(optarg);
-            if (!opts->diag_file_name) {
                 perror("strdup");
                 return (-1);
             }
@@ -370,7 +358,6 @@ static void usage(void)
             "\t-p <bake pool> - existing pool created with bake-mkpool\n"
             "\t[-c concurrency] - number of concurrent operations to issue "
             "with ULTs\n"
-            "\t[-d filename] - enable diagnostics output\n"
             "\t[-t client_progress_timeout,server_progress_timeout] # use \"-t "
             "0,0\" to busy spin\n"
             "\t[-r rpc_execution_streams] - number of ESs Margo should use for "

--- a/perf-regression/margo-p2p-bw.c
+++ b/perf-regression/margo-p2p-bw.c
@@ -41,7 +41,6 @@ struct options {
     char*         mmap_filename;
     unsigned int  mercury_timeout_client;
     unsigned int  mercury_timeout_server;
-    char*         diag_file_name;
     char*         na_transport;
     unsigned long g_buffer_size;
     int           align_buffer;
@@ -168,8 +167,6 @@ int main(int argc, char** argv)
     /* actually start margo */
     mid = margo_init_ext(g_opts.na_transport, MARGO_SERVER_MODE, &mii);
     assert(mid);
-
-    if (g_opts.diag_file_name) margo_diag_start(mid);
 
     /* adjust mercury timeout in Margo if requested */
     if (my_mpi_rank == 0 && g_opts.mercury_timeout_server != UINT_MAX) {
@@ -320,8 +317,6 @@ int main(int argc, char** argv)
     assert(ret == SSG_SUCCESS);
     ssg_finalize();
 
-    if (g_opts.diag_file_name) margo_diag_dump(mid, g_opts.diag_file_name, 1);
-
     free(gid_buffer);
 
     if (g_opts.mmap_filename == NULL) {
@@ -353,15 +348,8 @@ static int parse_args(int argc, char** argv, struct options* opts)
     /* warm up for 1 second by default */
     opts->warmup_seconds = 1;
 
-    while ((opt = getopt(argc, argv, "n:x:c:T:d:t:D:m:X:aw:")) != -1) {
+    while ((opt = getopt(argc, argv, "n:x:c:T:t:D:m:X:aw:")) != -1) {
         switch (opt) {
-        case 'd':
-            opts->diag_file_name = strdup(optarg);
-            if (!opts->diag_file_name) {
-                perror("strdup");
-                return (-1);
-            }
-            break;
         case 'x':
             ret = sscanf(optarg, "%d", &opts->xfer_size);
             if (ret != 1) return (-1);
@@ -435,7 +423,6 @@ static void usage(void)
         "ULTs\n"
         "\t[-T <os threads>] - number of dedicated operating system threads to "
         "run ULTs on\n"
-        "\t[-d <filename>] - enable diagnostics output\n"
         "\t[-t <client_progress_timeout,server_progress_timeout>] # use \"-t "
         "0,0\" to busy spin\n"
         "\t[-m <filename>] - use memory-mapped file as buffers instead of "

--- a/perf-regression/margo-p2p-latency.c
+++ b/perf-regression/margo-p2p-latency.c
@@ -25,7 +25,6 @@ struct options {
     int          iterations;
     unsigned int mercury_timeout_client;
     unsigned int mercury_timeout_server;
-    char*        diag_file_name;
     char*        na_transport;
     int          warmup_iterations;
     int          margo_forward_timed_flag;
@@ -110,8 +109,6 @@ int main(int argc, char** argv)
     /* actually start margo */
     mid = margo_init_ext(g_opts.na_transport, MARGO_SERVER_MODE, &mii);
     assert(mid);
-
-    if (g_opts.diag_file_name) margo_diag_start(mid);
 
     /* adjust mercury timeout in Margo if requested */
     if (my_mpi_rank == 0 && g_opts.mercury_timeout_server != UINT_MAX) {
@@ -202,8 +199,6 @@ int main(int argc, char** argv)
 
     ssg_finalize();
 
-    if (g_opts.diag_file_name) margo_diag_dump(mid, g_opts.diag_file_name, 1);
-
     free(gid_buffer);
 
     margo_finalize(mid);
@@ -226,17 +221,10 @@ static int parse_args(int argc, char** argv, struct options* opts)
     /* unless otherwise specified, do 100 iterations before timing */
     opts->warmup_iterations = 100;
 
-    while ((opt = getopt(argc, argv, "n:i:d:t:w:Tx:")) != -1) {
+    while ((opt = getopt(argc, argv, "n:i:t:w:Tx:")) != -1) {
         switch (opt) {
         case 'T':
             opts->margo_forward_timed_flag = 1;
-            break;
-        case 'd':
-            opts->diag_file_name = strdup(optarg);
-            if (!opts->diag_file_name) {
-                perror("strdup");
-                return (-1);
-            }
             break;
         case 'i':
             ret = sscanf(optarg, "%d", &opts->iterations);
@@ -284,7 +272,6 @@ static void usage(void)
         "\t-n <na> - na transport\n"
         "\t[-x <xfer_size>] - size of req/resp payload in bytes (defaults to "
         "0)\n"
-        "\t[-d filename] - enable diagnostics output\n"
         "\t[-t client_progress_timeout,server_progress_timeout] # use \"-t "
         "0,0\" to busy spin\n"
         "\t[-w <warmup_iterations>] - number of warmup iterations before "

--- a/perf-regression/margo-p2p-vector.c
+++ b/perf-regression/margo-p2p-vector.c
@@ -51,7 +51,6 @@ struct options {
     int           threads;
     unsigned int  mercury_timeout_client;
     unsigned int  mercury_timeout_server;
-    char*         diag_file_name;
     char*         na_transport;
     int           align_buffer;
     int           warmup_seconds;
@@ -164,8 +163,6 @@ int main(int argc, char** argv)
     /* actually start margo */
     mid = margo_init_ext(g_opts.na_transport, MARGO_SERVER_MODE, &mii);
     assert(mid);
-
-    if (g_opts.diag_file_name) margo_diag_start(mid);
 
     /* adjust mercury timeout in Margo if requested */
     if (my_mpi_rank == 0 && g_opts.mercury_timeout_server != UINT_MAX) {
@@ -334,8 +331,6 @@ int main(int argc, char** argv)
     assert(ret == SSG_SUCCESS);
     ssg_finalize();
 
-    if (g_opts.diag_file_name) margo_diag_dump(mid, g_opts.diag_file_name, 1);
-
     free(gid_buffer);
 
     free(g_buffer);
@@ -362,15 +357,8 @@ static int parse_args(int argc, char** argv, struct options* opts)
     /* warm up for 1 second by default */
     opts->warmup_seconds = 1;
 
-    while ((opt = getopt(argc, argv, "n:x:c:T:d:t:D:aw:v:")) != -1) {
+    while ((opt = getopt(argc, argv, "n:x:c:T:t:D:aw:v:")) != -1) {
         switch (opt) {
-        case 'd':
-            opts->diag_file_name = strdup(optarg);
-            if (!opts->diag_file_name) {
-                perror("strdup");
-                return (-1);
-            }
-            break;
         case 'x':
             ret = sscanf(optarg, "%lu", &opts->xfer_size);
             if (ret != 1) return (-1);
@@ -432,7 +420,8 @@ static void usage(void)
         stderr,
         "Usage: "
         "margo-p2p-bw -x <xfer_size> -D <duration> -n <na>\n"
-        "\t-x <xfer_size> - size of each bulk transfer in bytes, must be evenly "
+        "\t-x <xfer_size> - size of each bulk transfer in bytes, must be "
+        "evenly "
         "divisible by vector len\n"
         "\t-D <duration> - duration of test in seconds\n"
         "\t-n <na> - na transport\n"
@@ -441,7 +430,6 @@ static void usage(void)
         "ULTs\n"
         "\t[-T <os threads>] - number of dedicated operating system threads to "
         "run ULTs on\n"
-        "\t[-d <filename>] - enable diagnostics output\n"
         "\t[-t <client_progress_timeout,server_progress_timeout>] # use \"-t "
         "0,0\" to busy spin\n"
         "\t[-a] - explicitly align memory buffer to page size\n"

